py/terra_widgets/workspace_metadata.py

```diff
@@ -0,0 +1,79 @@
+"""Methods to obtain workspace metadata for the current user in various formats."""
+
+import json
+import os
+from typing import Dict
+
+import firecloud.api as fapi
+from IPython import get_ipython
+
+AOU_DOMAIN = '@researchallofus.org'
+EDIT_ACCESS_LEVELS = ['WRITER', 'OWNER', 'PROJECT_OWNER']
+
+
+class WorkspaceMetadata:
+  """Encapsulate all logic for obtaining workspace metadata."""
+
+  def __init__(self):
+    self.user = os.getenv('OWNER_EMAIL')
+    self.terra_workspaces = fapi.list_workspaces().json()
+    if self.user.endswith(AOU_DOMAIN):
+      # Use the All of Us API to get the human-readable workspace names. For All of Us workspaces,
+      # the Terra workspace metadata the workspace names are actually the AoU workspace ids.
+      aou_response = get_ipython().getoutput('''curl -H "Content-Type: application/json" \
```

Comment:
hmm.. we're don't officially support this - I will add as a discussion item on our eng sync tomorrow. Probably fine temporarily at least, but we'll need to make sure this is future proofed

Reply:
sg, please feel free to send a followup pull request to update it later when there is a better way